commons/ava_services/gecko_service_availability_checker_core.go

```diff
@@ -18,20 +21,17 @@ func (g GeckoServiceAvailabilityCheckerCore) IsServiceUp(toCheck services.Servic
 	client := gecko_client.NewGeckoClient(jsonRpcSocket.GetIpAddr(), jsonRpcSocket.GetPort())
 	healthInfo, err := client.HealthApi().GetLiveness()
 	if err != nil {
-		logrus.Trace(stacktrace.Propagate(err, "Error occurred in getting liveness info"))
+		logrus.Trace(stacktrace.Propagate(err, "Error occurred getting liveness info"))
 		return false
 	}
 
-	// HACK HACK HACK we need to wait for bootstrapping to finish, and there is not API for this yet (in development)
-	// TODO TODO TODO once bootstrapping checker is available, use that instead of just waiting
 	if healthInfo.Healthy {
-		time.Sleep(15 * time.Second)
+		time.Sleep(time.Second)
```

Comment:
Hey @aaronbuchwald , I'd prefer to not remove the TODO/comment/15-second sleep until we know for sure that this problem is fixed - it was causing a lot of nondeterminism in the past, and I don't think we have the readiness API available yet

Reply:
Switching back for now, but this is fixed on denali now